src/System.IO/tests/BinaryWriter/BinaryWriterTests.cs

```
@@ -0,0 +1,391 @@
+            }
+            dw2.Dispose();
+            mstr.Dispose();
+
+            // [] ArgumentException for invalid seekOrigin
+            mstr = new MemoryStream();
+            dw2 = new BinaryWriter(mstr);
+            dw2.Write("012345789".ToCharArray());
+            Assert.Throws<ArgumentException>(() => { lReturn = dw2.Seek(3, ~SeekOrigin.Begin); }, "Error_109ux! Expected exception not thrown");
+
+            dw2.Dispose();
+            mstr.Dispose();
+        }
+
+        [TestMethod]
+        public static void BinaryWriter_BaseStreamTests()
+        {
+            // [] Get the base stream for MemoryStream
+            MemoryStream ms2 = new MemoryStream();
+            BinaryWriter sr2 = new BinaryWriter(ms2);
+            if (sr2.BaseStream != ms2)
+                Assert.Fail("Error_g57hb! Incorrect basestream");
+
+            sr2.Dispose();
+            ms2.Dispose();
+        }
+
+        [TestMethod]
+        public static void BinaryWriter_FlushTests()
+        {
+            // [] Check that flush updates the underlying stream
+            MemoryStream memstr2 = new MemoryStream();
+            BinaryWriter bw2 = new BinaryWriter(memstr2);
+
+            bw2.Write("HelloWorld");
+            //TODO:: Ckeck with dev why it's 11 bytes.
+            Assert.AreEqual(11, memstr2.Length, "Error_984y7! Incorrect stream length==" + memstr2.Length);
+            bw2.Flush();
+            Assert.AreEqual(11, memstr2.Length, "Error_8589v! Incorrect stream length after flush==" + memstr2.Length);
+
+            // [] Flushing closed writer should not throw an exception.
+            memstr2 = new MemoryStream();
+            bw2 = new BinaryWriter(memstr2);
+            bw2.Dispose();
+            try
+            {
+                bw2.Flush();
+            }
+            catch (Exception exc)
+            {
+                Assert.Fail("Error_399c;! Unexpected exception exception thrown, exc==" + exc.ToString());
```

Rather than catching an asserting, for cases like this we usually just let the exception propagate out of the test.